### ISSUE TITLE
Mel scaling

### ISSUE THREAD

**Issue by survirtual** [phip1611/spectrum-analyzer#44]
Thank you for providing this codebase.  Looking through the docs and code, I didn't see built-in support for generating a mel-scale frequency domain.  Is this correct?

**Comment by phip1611** [comment]
True. I never heard of the Mel scale to be honest

**Comment by phip1611** [comment]
Feel free to submit a PR

**Comment by survirtual** [comment]
Mel scale is used to generate a mel spectrogram, which is what nearly every machine learning model uses to analyze audio.
https://en.wikipedia.org/wiki/Mel_scale

I went ahead and implemented my own spectrogram code with mel scaling.  All you'd need to do here to add it to this lib is generate a mel filter (there is a decent crate that does this) then dot it with the FFT output.

I'm short on time but I might make a PR for your codebase; yours was the closest to what I needed before I implemented myself.
